TwitchPlaysAssembly/Src/ComponentSolvers/Modded/Shims/SwanShim.cs

```diff
@@ -23,6 +29,137 @@ protected override IEnumerator RespondToCommandShimmed(string inputCommand)
 			yield return "awardpoints 1";
 	}
 
+	protected override IEnumerator ForcedSolveIEnumeratorShimmed()
+	{
+		yield return null;
+
+		if (!_component.GetValue<bool>("solved") && _component.GetValue<bool>("executeLock"))
+		{
+			if (Unshimmed.ForcedSolveMethod == null) yield break;
+			var coroutine = (IEnumerator) Unshimmed.ForcedSolveMethod.Invoke(Unshimmed.CommandComponent, null);
+			while (coroutine.MoveNext())
+				yield return coroutine.Current;
+			yield break;
+		}
+		Module.BombComponent.StartCoroutine(HandleSolve());
+		while (!_component.GetValue<bool>("solved") || _component.GetValue<int>("timeUpCounter") <= 8) yield return true;
+	}
+
+	IEnumerator HandleSolve()
+	{
+		reset:
+		while (_component.GetValue<bool>("keyboardLock")) yield return null;
+		string input = _component.GetValue<string>("computerText");
+		string ans = GetAnswer(_component.GetValue<int>("systemResetCounter"), _component.GetValue<int>("solveRange") <= 0);
+		if (input.Length > ans.Length)
+		{
+			if (Unshimmed.ForcedSolveMethod == null) yield break;
+			var coroutine = (IEnumerator) Unshimmed.ForcedSolveMethod.Invoke(Unshimmed.CommandComponent, null);
+			while (coroutine.MoveNext())
+				yield return coroutine.Current;
+			yield break;
+		}
+		else
+		{
+			for (int i = 0; i < input.Length; i++)
+			{
+				if (i == ans.Length)
+					break;
+				else if (input[i] != ans[i])
+				{
+					if (Unshimmed.ForcedSolveMethod == null) yield break;
+					var coroutine = (IEnumerator) Unshimmed.ForcedSolveMethod.Invoke(Unshimmed.CommandComponent, null);
+					while (coroutine.MoveNext())
+						yield return coroutine.Current;
+					yield break;
+				}
+			}
+		}
+		if (ans == ">: 77")
+		{
+			ans = ">: 7 7 ";
+			if (input == ">: 77")
+				input = ">: 7 7 ";
+		}
+		ans = ans.Replace(">:", "").Trim();
+		input = input.Replace(">:", "").Trim();
```

Comment:
Wouldn't it be better to do this to `input` when you get it initially and then not have `GetAnswer()` include `>:`?

Reply:
The autosolver for this caused me many problems, so I didn't care about optimization at some point. I can look at it. Thanks for the comments again.